### PR TITLE
Fix completion sorting in remote method calls and checking expressions

### LINKED ISSUE #33624: Auto-completions spec conformance for the `CheckingExpression`
**Description:**
$title
https://ballerina.io/spec/lang/master/#checking-expr

### LINKED ISSUE #33625: Auto-completions spec conformance for the `RemoteMethodCallAction`
**Description:**
$title

### LINKED ISSUE #33672: Code completion for remote function can be sorted based on LHS of the expression
**Description:**
Screenshot 2021-11-05 at 21 02 16

Here sql:ProcedureCallResult[] is only returned for call method. This can be highlighted.

### PR DESCRIPTION
**Pull request by IMS94**
## Purpose
$subject

Fixes #33625
Fixes #33672
Fixes #33624

## Approach
Updated `ContextTypeResolver` and relevant completion providers to adjust the completion sorting.

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/completion/ActionNodeContextTest.java [hunk 1]
```diff
@@ -35,6 +35,10 @@ public void test(String config, String configPath) throws IOException, Workspace
     @Override
     public Object[][] dataProvider() {
         return this.getConfigsList();
+//        return new Object[][]{
```
Comment (malinthar) [inline language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/completion/ActionNodeContextTest.java:38]: Shall we remove these lines

**Review by malinthar — CHANGES_REQUESTED** [review]
Changes look good!

File: language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/completion/ActionNodeContextTest.java (hunk 1 above)
Comment (IMS94) [reply language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/completion/ActionNodeContextTest.java]: Done. It was a mistake.

**Comment by IMS94** [comment]
@nadeeshaan @malinthar @mohanvive With this PR, I have updated the assignability check as follows:
1. Items which are directly assignable to the context are sorted first
2. Items which are assignable after adding a `check` expression are sorted 2nd - This is similar to considering the `edit distance`. When we write code, we usually do `int varName = clientObj->methodCall()` and then add the `check` in between. That's why I thought of using this idea. Since remote method calls usually comes with an error union, this should be very useful.
3. Other items are sorted 3rd

WDYT?

**Comment by nadeeshaan** [comment]
> @nadeeshaan @malinthar @mohanvive With this PR, I have updated the assignability check as follows:
> 
> 1. Items which are directly assignable to the context are sorted first
> 2. Items which are assignable after adding a `check` expression are sorted 2nd - This is similar to considering the `edit distance`. When we write code, we usually do `int varName = clientObj->methodCall()` and then add the `check` in between. That's why I thought of using this idea. Since remote method calls usually comes with an error union, this should be very useful.
> 3. Other items are sorted 3rd
> 
> WDYT?

This is very good thinking @IMS94.  We do sort within the items related check as well right?

**Comment by IMS94** [comment]
> This is very good thinking @IMS94. We do sort within the items related check as well right?
@nadeeshaan yes, as of now we do default sorting within each group (mentioned above). Depending on the context, we can customize that as well.

**Review by nadeeshaan — APPROVED** [review]
LGTM

**Review by mohanvive — APPROVED** [review]

**Comment by mohanvive** [comment]
> Changes look good!

@malinthar Need your approval to merge the PR.

**Review by malinthar — APPROVED** [review]
LGTM!

**Pull request merged**